File: MvcTaskManger/MvcTaskManger/Models/Project.cs
using MvcTaskManger.ViewModels;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MvcTaskManger.Models
{
    public class Project
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.None)]
        public int ProjectID { get; set; }

        public string ProjectName { get; set; }

        public DateTime DateOfStart { get; set; }

        public int? TeamSize { get; set; }

        public bool Active { get; set; }

        public string Status { get; set; }

        public int ClientLocationID { get; set; }  // Foreign Key of ClientLocations table  

        [ForeignKey("ClientLocationID")]
        public virtual ClientLocation ClientLocation { get; set; }  // Navigation Property

    }

    public static class ProjectExtention
    {
        public static ProjectViewModel ConvertProjectToProjectViewModel(this Project project)
        {
            return new ProjectViewModel()
            {
                ProjectID = project.ProjectID,
                ProjectName = project.ProjectName,
                DateOfStart = project.DateOfStart.ToString(),
                TeamSize = project.TeamSize,
                Active = project.Active,
                Status = project.Status,
                ClientLocationID = project.ClientLocationID,
                ClientLocation = project.ClientLocation
            };
        }
    }


}

File: TaskManager-Angular/src/app/models/project.ts
import { ClientLocation } from "./client-location";

//  module class to represent the structure of Project table
export class Project {

    // Properties
    projectID: number | any;
    //id: any;  // Only for json-server
    projectName: string | any;
    dateOfStart: string | any;
    teamSize: number | any;
    active: boolean | any;
    status: string | any;
    clientLocationID: number | any;
    clientLocation: ClientLocation | any;

    constructor() {
        this.projectID = null;
        //this.id = 0;   // Only for json-server
        this.projectName = null;
        this.dateOfStart = null;
        this.teamSize = null;
        this.active = false;
        this.status = null;
        this.clientLocationID = null;
        this.clientLocation = null;
    }

}
